### PR TITLE
istiod-default-validator not respect custom istioNamespace

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

fix: https://github.com/istio/istio/issues/38153

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by zirain** [comment]
@Monkeyanator PTAL.

**Comment by zirain** [comment]
@Monkeyanator PTAL.

File: istioctl/pkg/tag/generate_test.go [hunk 1]
```diff
@@ -114,71 +114,90 @@ var (
...
-		webhook admit_v1.MutatingWebhookConfiguration
-		whURL   string
-		whSVC   string
-		whCA    string
+		name           string
+		istioNamespace string
+		webhook        admit_v1.MutatingWebhookConfiguration
+		whURL          string
+		whSVC          string
+		whSvcNamespace string
```
Comment (irisdingbj) [inline istioctl/pkg/tag/generate_test.go:122]: Is this a duplicate for istioNamespace?

**Review by irisdingbj — APPROVED** [review]
Thanks for fixing this!

**Comment by ericvn** [comment]
It looks like the cheery-pick of this PR is causing a regression in the 1.13 stream of the external control plane test. This PR was never cherry-picked to 1.14 and we don't see the failure there. Until we release again, we won't test against the main branch which is likely why we haven't seen the failure there.

**Pull request merged**